internal/master/service.go

```diff
@@ -42,18 +44,53 @@ type standaloneService struct {
 	lg         *zap.Logger
 	statsCli   stats.Client
 	regionInfo *serverpb.RegionInfo
+	isClosed bool
+	healthCheckPeriod uint8
 }
 
 func (ss *standaloneService) GetStatus(ctx context.Context, request *emptypb.Empty) (*serverpb.RegionInfo, error) {
 	return ss.regionInfo, nil
 }
 
+
+func(ss *standaloneService) Check(ctx context.Context, healthCheckReq *serverpb.HealthCheckRequest) (*serverpb.HealthCheckResponse, error) {
+	if !ss.isClosed && ss.regionInfo != nil && ss.regionInfo.Status == serverpb.RegionStatus_LEADER {
+		return &serverpb.HealthCheckResponse{Status: serverpb.HealthCheckResponse_SERVING}, nil
+	}
+	return &serverpb.HealthCheckResponse{Status: serverpb.HealthCheckResponse_NOT_SERVING}, nil
+}
+
+func (ss *standaloneService) Watch(req *serverpb.HealthCheckRequest, watcher serverpb.HealthCheck_WatchServer) error {
+	ticker := time.NewTicker(time.Duration(ss.healthCheckPeriod) * time.Second)
+	defer ticker.Stop()
+	for {
+		select {
+		case <-ticker.C:
+			if ss.isClosed {
+				break
+			}
+			res, _ := ss.Check(context.Background(), req)
+			if err := watcher.Send(res); err != nil {
+				return err
+			}
+		}
+		if ss.isClosed {
```

Comment:
Using a channel might be better way than this if check after a select.

Reply:
Moved to channel based implementation